lib/resque.rb

```diff
@@ -271,11 +271,23 @@ def push(queue, item)
     data_store.push_to_queue(queue,encode(item))
   end
 
-  # Pops a job off a queue. Queue name should be a string.
+  # Pops a job off a set of queues. Queue names should be a string.
   #
   # Returns a Ruby object.
-  def pop(queue)
-    decode(data_store.pop_from_queue(queue))
+  def pop(*queues)
+    queue, decoded_job = pop_with_queue(*queues)
+
+    decoded_job
+  end
+
+  # Pops a job off a set of queues. Queue names should be strings.
+  #
+  # Returns an array of [queue, Ruby object] or nil.
+  def pop_with_queue(*queues)
```

Comment:
Since this is a newly introduced method, we could use the plural in the name. We could also use `from` instead of `with`.

```suggestion
  def pop_from_queues(*queues)
```

Reply:
What's different with `pop_with_queue()` is that it returns a pair of `[queue, job]` instead of just the `job` like `pop()` does, thus the `_with_queue` suffix. It's not that clear though.